index.bs

```diff
@@ -1160,4298 +1160,4345 @@ access to that data in a {{Window}} global.
-    ? value: text,
-    ? binaryValue: [ uint ]
-    domain: text,
-    path: text,
-    ? expires: js-uint,
-    size: js-uint,
-    httpOnly: bool,
-    secure: bool,
-    sameSite: "strict" / "lax" / "none",
-};
-</pre>
+      1. Return [=success=] with data [=constructor/Map=](|iterable|).
 
-The <code>network.Cookie</code> type represents a cookie.
+    <dt>|type| is the string "<code>object</code>"
+    <dd>
+      1. Let |deserialized key-value list| be a result of [=trying=] to
+         [=deserialize key-value list=] with |value|, |realm| and |session|.
 
-If the cookie value can be represented as a UTF-8 encoded string, the
-<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
-field will be present and consist of an array of integers representing the bytes
-of the cookie value.
+      1. Let |iterable| be [=CreateArrayFromList=](|deserialized key-value list|)
 
-<div algorithm>
-To <dfn>get a cookie</dfn> given |stored cookie|:
+      1. Return [=success=] with data [=Object.fromEntries=](|iterable|).
 
-Note: The definitions of |stored cookie|'s fields are from [[COOKIES]], except
-samesite-flag, which is from [[SAME-SITE-COOKIES]].
+    <dt>|type| is the string "<code>regexp</code>"
+    <dd>
+      1. Let |pattern| be the value of the <code>pattern</code> field of |local protocol value|.
 
-1. Let |name| be the result of [=UTF-8 decode=] with |stored cookie|'s name field.
+      1. Let |flags| be the value of the <code>flags</code> field of |local protocol value| or
+         undefined if no such a field.
 
-1. Let |utf8 decoded value| be the result of [=UTF-8 decode without BOM or fail=]
-   with |stored cookie|'s value.
+      1. Let |regex_result| be [=Regexp=](|pattern|, |flags|). If this throws exception, return [=error=]
+         with [=error code=] [=invalid argument=].
 
-1. If |utf8 decoded value| is failure, then:
+      1. Return [=success=] with data |regex_result|.
 
-    1. Let |value| be null and |binary value| be an empty list.
+    <dt>|type| is the string "<code>set</code>"
+    <dd>
+      1. Let |deserialized value list| be a result of [=trying=] to [=deserialize value list=]
+         given |value|, |realm| and |session|.
 
-    1. For each |byte| in |stored cookie|'s value:
+      1. Let |iterable| be [=CreateArrayFromList=](|deserialized key-value list|)
 
-      1. Append the [=byte/value=] of |byte| to |binary value|.
+      1. Return [=success=] with data [=Set object=](|iterable|).
 
-   Otherwise: Let |value| be |utf8 decoded value| and |binary value| be null.
+    <dt>otherwise
+    <dd>Return [=error=] with [=error code=] [=invalid argument=].
+  </dl>
 
-1. Let |domain| be |stored cookie|'s domain field.
+</div>
 
-1. Let |path| be |stored cookie|'s path field.
 
-1. Let |expires| be |stored cookie|'s expires field represented as a unix
-   timestamp, if set, or null otherwise.
+#### The script.PrimitiveProtocolValue Type #### {#type-script-PrimitiveProtocolValue}
```

Comment:
nit: alphabetical order (if we care about this here...)
- script.PrimitiveProtocolValue should be after script.PreloadScript
- script.ResultOwnership should be before script.SharedId